tools/preflight/helper.go

```diff
@@ -0,0 +1,448 @@
+							"cpu":    resource.MustParse("250m"),
+						},
+						Limits: map[corev1.ResourceName]resource.Quantity{
+							"memory": *resource.NewQuantity(128*1024*1024, resource.BinarySI),
+							"cpu":    resource.MustParse("500m"),
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyAlways,
+		},
+	}
+
+	if op.ImagePullSecret != "" {
+		pod.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
+			{
+				Name: op.ImagePullSecret,
+			},
+		}
+	}
+	if op.ServiceAccountName != "" {
+		pod.Spec.ServiceAccountName = op.ServiceAccountName
+	}
+
+	return pod
+}
+
+func createVolumeSnapshotPVCSpec(storageClass, namespace string) *corev1.PersistentVolumeClaim {
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourcePvc + resNameSuffix,
+			Namespace: namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			StorageClassName: &storageClass,
+			Resources: corev1.ResourceRequirements{
+				Requests: map[corev1.ResourceName]resource.Quantity{
+					"storage": resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+
+	return pvc
+}
+
+func createVolumeSnapshotPodSpec(namespace, pvcName string) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourcePod + resNameSuffix,
+			Namespace: namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "busybox",
+					Image:   "busybox",
+					Command: []string{"bin/sh", "-c"},
+					Args:    []string{"touch /demo/data/sample-file.txt && sleep 3000"},
+					Resources: corev1.ResourceRequirements{
+						Requests: map[corev1.ResourceName]resource.Quantity{
+							"memory": resource.MustParse("64Mi"),
+							"cpu":    resource.MustParse("250m"),
+						},
+						Limits: map[corev1.ResourceName]resource.Quantity{
+							"memory": resource.MustParse("128Mi"),
+							"cpu":    resource.MustParse("500m"),
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "source-data",
+							MountPath: "/demo/data",
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "source-data",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+							ReadOnly:  false,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return pod
+}
+
+// createVolumeSnapsotSpec creates pvc for volume snapshot
+func createVolumeSnapsotSpec(name, namespace, snapVer, pvcName string) *unstructured.Unstructured {
```

Comment:
refer backup controller code .. check how it creates volume snapshot spec as per version.. reuse it